frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -721,7 +774,21 @@ function ReaderRolling:onGotoViewRel(diff)
         else
             diff = math.floor(diff)
         end
-        self:_gotoPage(self.current_page + diff*page_count)
+        local new_page = self.current_page
+        if self.ui.document:hasHiddenFlows() then
+            local test_page = 0
+            for i=1, math.abs(diff*page_count) do
+                if diff > 0 then
+                    test_page = self.ui.document:getNextPage(new_page)
+                else
+                    test_page = self.ui.document:getPrevPage(new_page)
+                end
+                if test_page > 0 then new_page = test_page end
```

Comment:
3 lines please (might be a question of taste, but there I was looking for new_page getting updated and didn't see it at first).